packages/nodejs/lib-os.js.patch

```diff
@@ -1,7 +1,25 @@
-diff '--color=auto' -uNr node-v17.8.0.orig/lib/os.js node-v17.8.0/lib/os.js
---- node-v17.8.0.orig/lib/os.js	2022-03-26 22:11:04.145459285 +0530
-+++ node-v17.8.0/lib/os.js	2022-03-26 22:12:19.585459257 +0530
-@@ -183,7 +183,7 @@
+--- ./lib/os.js.orig	2022-05-21 13:09:26.332384353 +0530
++++ ./lib/os.js	2022-05-21 13:10:15.732384334 +0530
+@@ -45,6 +45,7 @@
+ const { validateInt32 } = require('internal/validators');
+ 
+ const {
++  getNumCPUs,
+   getCPUs,
+   getFreeMem,
+   getHomeDirectory: _getHomeDirectory,
+@@ -114,6 +115,10 @@
+   return [avgValues[0], avgValues[1], avgValues[2]];
+ }
+ 
++function numcpus() {
```

Comment:
This is a ugly fix for concurrency deadlock when passing a directory to `node --test`. See `src-node_os.cc.patch` and `lib-os.js.patch` for more information. This is a **non-standard** function, relying on this is bad. libuv 1.44 seems to have a `uv_available_parallelism` which Node.js is willing to adopt for getting concurrency limits as seen in lib/internal/test_runner/test.js:34